src/database/adatabase.cpp

```diff
@@ -33,14 +33,33 @@ QString ADatabaseError::text() const
 
 ADatabase* ADatabase::instance = nullptr;
 
+//[F]: This is what I mentioned the other day, instead of writing the table names
+// to a member variable once at connect(), we are accessing the layout at the moment we query it.
+// While being a little more expensive, this makes the app much more robust because
+// it always returns an accurate description of the database, regardless of whether
+// it is already created or not. Also it encapsulates this functionality here instead
+// of being included in connect()
+/*!
+ * \brief Return the names of the tables in the database.
+ */
 TableNames ADatabase::getTableNames() const
 {
-    return tableNames;
+    auto db = ADatabase::database();
+    return db.tables();
 }
```

Comment:
While this needs benchmarking to determine if its better. 
At a glance i see it as worse. I dont know how db.tables() works in the background but re-reading the tables with each query doesnt seem efficient.
reading them upfront and simply updating them seems more reasonable. 

Once the program starts the database is read, the local data gets populated for quicker access.
Any updates to the db update the local data and at the end the sql database is edited.

Mechanically an accessor can remove the clunky tableData.value(...)
I assume we will not see any preformance difference right now, so we should consider setting up a testing module.
Im up for it if its not particularly interesting for you.

Reply:
It is a bit more expensive performance-wise, yes. But it adds value by being more robust. Also it only affects the commit queries, so the majority of queries are unaffected by this. Accessing data from the database is very fast, so I am not concerned about the performance loss considering that we are gaining always up-to-date data instead of keeping a (potentially out-of-date) copy of that data around in a member variable.

Reply:
thinking it through again you make a good point though, the changes in layout really only happen during the initial setup so taking the performance hit throughout the application seems unreasonable. Implementing an updateLayout() function to update the member variables whenever layout changes.

Reply:
That does seem like a good hybrid in our case at least we *prove* otherwise

Reply:
done